### ISSUE TITLE
Does not mention encoding requirements

### ISSUE THREAD

**Issue by oskarb** [MicrosoftDocs/sql-docs#7228]
Excerpt:

>  leveraging an intelligent framework known as Program Synthesis using Examples (PROSE). For a user without specialized domain knowledge, importing data can often be a complex, error prone, and tedious task. This wizard streamlines the import process as simple as selecting an input file and unique table name, and the PROSE framework handles the rest.

I.e. automagic this and automagic that. But it cannot do a simple text encoding detection which any random editor can do? CSV in Windows-1252 results in Swedish characters being mangled. CSV in UTF-8 also results in Swedish characters being mangled.

Not only does it not detect the encoding - it also offers me no control to *select* encoding, nor is there anything in the tool or on the linked documentation page to *tell* me what encoding must be used.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d89ecc7f-4705-dabe-9f70-2070d4aa19c9
* Version Independent ID: 9c991bd5-535c-03b5-d810-c1b2ca7e7cb4
* Content: [Import Flat File to SQL - SQL Server](https://docs.microsoft.com/en-us/sql/relational-databases/import-export/import-flat-file-wizard?view=sql-server-ver15#troubleshooting)
* Content Source: [docs/relational-databases/import-export/import-flat-file-wizard.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/import-export/import-flat-file-wizard.md)
* Product: **sql**
* Technology: **data-movement**
* GitHub Login: @yualan
* Microsoft Alias: **alayu**

**Comment by WilliamAntonRohm** [comment]
@oskarb -- Oskar, thank you for your feedback.  Please consider these resources:
- [SQL Server on Microsoft Q&A](https://docs.microsoft.com/answers/products/sql-server)
- [DBA Stack Exchange](https://dba.stackexchange.com/questions/tagged/sql-server)
- [Stack Overflow](https://stackoverflow.com/questions/tagged/sql-server)

If you feel this is a feature issue, please add it here:
- [SQL Server feedback](http://aka.ms/sqlfeedback)

@yualan -- Alan, please look into this potential documentation issue.

**Comment by erinstellato-ms** [comment]
Hi @oskarb - I checked with engineering and we created an issue internally to try and improve this.  It likely won't be until the next major release of SSMS (SSMS 20), but are planning to make updates to make this better with regard to encoding.  For the moment, I have added a note to the docs page you referenced under troubleshooting with a bit more info about expected encoding based on the system

**Comment by oskarb** [comment]
Thanks for getting back to this.

I have some concerns.

> Currently, the importer uses encoding based on the system's active code page. On most machines this defaults to ANSI.

As I understand it, "ANSI" in the context of Windows doesn't mean any specific encoding, just "the active code page" and in some programs (e.g. Notepad) it used to mean "not unicode". So I guess that's a clue, yet I'm not fully convinced.

First, what is the active code page? I found two suggestions for how to check and they produce different answers:

![image](https://github.com/MicrosoftDocs/sql-docs/assets/482095/1eacb7b5-a5ab-433f-b8e3-87b4aa9adb6e)

It's been a year and a half since I opened this topic, but it's the same machine and I see no reason for why these settings should have changed. Yet when I previously tried with a file explicitly encoded in Windows-1252, it did not work for some characters. I did not get the idea to try with the DOS code page 437.

Since opening this issue I later finalized an internal KB-article to document a procedure where using "Import flat file" was one of the steps. In this article I have noted that UTF-16 works for our purposes (i.e. involving Swedish character like åäö). With UTF-16 always using 2 bytes per character for the relevant set of characters, it could not work if it was accidentally read as CP437 or CP1252

It seems likely that the file started with a UTF-16 byte order mark. So I am assuming that something in the importer does in fact look for that and is capable of then using UTF-16.  This doesn't explain why I had trouble with Windows-1252  despite my machine indicating Windows-1252 being the default.
